plugins/zns/zns.c

```diff
@@ -223,7 +223,20 @@ static int zone_mgmt_send(int argc, char **argv, struct command *cmd, struct plu
 	if (fd < 0)
 		return errno;
 
-	if (cfg.data_len) {
+	if (!cfg.zsa) {
+		fprintf(stderr, "zone send action must be specified\n");
+		err = -EINVAL;
+		goto close_fd;
+	}
+
+	if (cfg.zsa == 0x10){
+		if(!cfg.data_len){
+			cfg.data_len = get_zdes_bytes(fd, cfg.namespace_id);
```

Comment:
If zdes = 0, then the command returns a status of invalid field in command. 

One could stop here without going further and allocate buffers (of zero size).

Reply:
@MatiasBjorling 
Great Idea